jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientTest.java

```diff
@@ -79,6 +80,158 @@
 
 public class HttpClientTest extends AbstractTest
 {
+    @ParameterizedTest
+    @MethodSource("transports")
+    public void testClientUseContentSourceInSpawnedThreadEmptyResponseContent(Transport transport) throws Exception
+    {
+        start(transport, new Handler.Abstract()
+        {
+            @Override
+            public boolean handle(Request request, org.eclipse.jetty.server.Response response, Callback callback)
+            {
+                response.write(true, BufferUtil.EMPTY_BUFFER, callback);
+                return true;
+            }
+        });
+
+        var response = new Response.Listener()
+        {
+            final CompletableFuture<String> body = new CompletableFuture<>();
+
+            @Override
+            public void onContentSource(Response response, Content.Source contentSource)
+            {
+                new Thread(() ->
+                {
+                    Content.Chunk chunk = contentSource.read();
+                    if (chunk == null)
+                    {
+                        contentSource.demand(() -> onContentSource(response, contentSource));
+                        return;
+                    }
+
+                    chunk.release();
+
+                    if (!chunk.isLast())
+                        contentSource.demand(() -> onContentSource(response, contentSource));
+                    else
+                        body.complete("");
+                }
+                ).start();
+            }
+        };
+
+        client.newRequest(server.getURI())
+            .method("POST")
+            .send(response);
```

Comment:
We need to parametrize this test with zero content, but also some content.

Reply:
This is done in another test, I've refactored the newly added ones to make them clearer.